### PR TITLE
Add session renewal, session listing

### PR DESCRIPTION
**Pull request by kim-toms**
I added the functions and tests for session renewal and the 'list' function for sessions.  If you'd like something additional, let me know.

### CODE REVIEW

File: lib/diplomat/session.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 module Diplomat
   class Session < Diplomat::RestClient
 
-    @access_methods = [ :create, :destroy ]
+    @access_methods = [ :create, :destroy, :list ]
```
Comment (johnhamelink) [inline lib/diplomat/session.rb]: Could you add renew to this list so it's accessible from `Diplomat::Session.list(id)`?

**Comment by johnhamelink** [comment]
Thanks @kim-toms! It looks almost perfect, except for adding in that list item to `@access_methods`.

Thanks for contributing!

Best,
JH

File: lib/diplomat/session.rb [hunk 2]
```diff
@@ -28,5 +28,25 @@ def destroy id
...
       return raw.body
     end
+    
+    # List sessions
+    # @return [Struct]
+    def list
+      raw = @conn.get do |req|
+        req.url "/v1/session/list/"
+      end
+      return JSON.parse(raw.body)
```
Comment (mcansky) [inline lib/diplomat/session.rb]: you don't need the return

File: lib/diplomat/session.rb [hunk 3]
```diff
@@ -28,5 +28,25 @@ def destroy id
...
+    end
+    
+    # Renew session
+    # @param id [String] session id
+    # @return [Struct]
+    
+    def renew id
+      raw=@conn.put do |req|
```
Comment (mcansky) [inline lib/diplomat/session.rb]: syntax : be consistent

File: spec/session_spec.rb [hunk 4]
```diff
@@ -28,6 +28,26 @@
       expect(session.destroy("fc5ca01a-c317-39ea-05e8-221da00d3a12")).to eq("true")
     end
 
+    it "list" do
+      json = JSON.generate(
```
Comment (mcansky) [inline spec/session_spec.rb]: use let for this kind of things

File: spec/session_spec.rb [hunk 5]
```diff
@@ -28,6 +28,26 @@
...
+        [{"ID" => "fc5ca01a-c317-39ea-05e8-221da00d3a12"}]
+      )
+
+      faraday.stub(:get).and_return(OpenStruct.new({body: json}))
```
Comment (mcansky) [inline spec/session_spec.rb]: that's a setup thing, a before loop is more appropriate to host it

**Comment by johnhamelink** [comment]
Hey @kim-toms - are you going to have a chance to fix the issues @mcansky and I pointed out, I'd like to merge this :smile:

**Comment by kim-toms** [comment]
Yes, sorry, I've gotten pre-occupied.  I'll get to it tomorrow.  I was a
little confused by some of the advice, which is what led to the delay.

**Comment by johnhamelink** [comment]
No problem @kim-toms, looking forward to hearing back from you :smile:

**Comment by johnhamelink** [comment]
Thanks @kim-toms. Released in [0.11.0](https://rubygems.org/gems/diplomat/versions/0.11.0)

**Pull request merged**